enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
Each init-container must have unique names as the Deployment becomes unusable if they have 2 init-containers with the same name - the whole Deployment is rejected by kubernetes in the case of duplicate names. 

Having actual duplicates of the same container metadata with different names may or may not break things - doing a chmod twice is fine but other init stuff may barf if done twice. And yes - ignoring a different init-container that just happens to have the same name will lose init-container logic that will probably break things as well. (Though at least the Deployment won't be rejected by kubernetes - but something bad will most likely happen down the line).

Given the plugin & configurable nature of fmp it'd be nice to try nail the logic going forwards as I can see lots of different developers writing lots of generators+enrichers+custom metadata of various shapes and sizes for different use cases & frameworks and with ordering and whatnot I'm sure clashes will happen. Probably all possible clashes will happen at some point ;) (The same kinds of clashes can happen all over the place TBH with services/ports/containers of the same name too).

So maybe we should:

* if the containers are real duplicates log a warning; we could maybe even get clever and tell the developer that its probably an ordering thingy fix on the enricher (e.g. your patch to the volume enricher to avoid dupes) to help them really fix the problem properly. This warning in strict mode could fail the build as well (for CI jobs)
* if the containers are different but the names are the same we could either log a warning and change the name of the 2nd one automatically to something unique (which is a kind option for downstream projects; trying to work around mistakes/regressions in fmp or fmp's configuration and keeping the lights on - and the init container name is fairly meaningless in the grand scheme of things anyways; adding a postfix to enforce uniqueness sounds OK to me as a temporary work around) or fail the build until the a new release of fmp comes out with updated enrichers or the user figures out a manual workaround